sdk/search/azure-search/README.md

```diff
@@ -0,0 +1,211 @@
+# Azure cognitive search client library for Java
+
+This is the Java client library for [Azure Cognitive Search](https://docs.microsoft.com/en-us/rest/api/searchservice/).
+Azure Cognitive Search is a fully managed cloud search service that provides a rich search experience to custom applications.
+This library provides an easy (native) way for a Java developer to interact with the service to accomplish tasks like: create and manage indexes, load data, implement search features, execute queries, and handle results.
+
+[Source code][source_code] | [Package (Maven)][package] | [API reference documentation][api_documentation]
+| [Product documentation][search_docs] | [Samples][samples]
+
+## Getting started
+
+### Prerequisites
+
+-   Java Development Kit (JDK) with version 8 or above
+-   [Azure subscription][azure_subscription]
+-   An instance of [Cognitive search service][search]
+
+### Adding the package to your product
+
+[//]: # ({x-version-update-start;com.azure:azure-search;current})
+```
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-search</artifactId>
+    <version>1.0.0-beta.2</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+## Using the library
+
+The APIs documented in this section provide access to operations on search data, such as index creation and population, document upload, and queries. When calling APIs, keep the following points in mind:
+
+-   All APIs will be issued over HTTPS (on the default port 443).
+-   Your search service is uniquely identified by a fully-qualified domain name (for example: `mysearchservice.search.windows.net`).
+-   All requests must include an `api-key` that was generated for the Search service you provisioned. Having a valid key establishes trust, on a per request basis, between the application sending the request and the service that handles it.
+-   Requests can optionally specify an `api-version`. The default value matches the version of the current service release: `2019-05-06`.
+-   You will need to provide the `cognitive search service url` which is the URL of the Azure Cognitive Search service you provisioned: `https://<yourService>.search.windows.net`.
+
+## Key concepts
+
+Azure Cognitive Search has the concepts of search services and indexes and documents, where a search service contains one or more indexes that provides persistent storage of searchable data, and data is loaded in the form of JSON documents. Data is typically pushed to an index from an external data source, but if you use an indexer, it's possible to crawl a data source to extract and load data into an index.
+
+There are several types of operations that can be executed against the service:
+
+-   [Index management operations](https://docs.microsoft.com/en-us/rest/api/searchservice/index-operations). Create, delete, update, or configure a search index.
```

Comment:
@brjohnstmsft I can compltely remove this part and point to it in a link. Other SDKs do have some ket concept overview on the service. This can always be updated.
@alzimmermsft what is the best approach here?